### ISSUE TITLE
Check cloud vendor availability

### ISSUE THREAD

**Issue by KazuCocoa** [appium/appium-inspector#128]
Appium Inspector has a feature to connect to cloud vendor environments. Each vendor contributed to AppiumDesktop to allow AD to attach to their environment.

This repository (appium-inspector) now uses [web2driver](https://github.com/projectxyzio/web2driver), which uses webdriverio as its backend, to support W3C spec capabilities for Appium 2.0. Appium 2.0 will drop MJSONWP style, so this change is mandatory as appium inspector.

Then, potentially vendors have an issue in their connectivity. I guess especially vendors modify `desiredCapabilities` might affect them.

**Comment by KazuCocoa** [comment]
A note about which vendor is still checking.

- [ ] SauceLabs: https://github.com/appium/appium-inspector/issues/22
- [x] HeadSpin
- [ ] Perfecto
- [ ] BrowserStack
- [ ] BitBar
- [ ] Kobiton
- [ ] pCloudy
- [ ] TestingBot
- [ ] ExperiTest
- [ ] Roboticmobi

cc @jlipps

**Comment by KazuCocoa** [comment]
Hello everyone. Thanks for adding your vendor availability in AppiumDesktop.
AppiumDesktop is moving to appium-inspector (inspector part) and appium-desktop (bundled appium server part). This repository is the new inspector part.

The backend is going to W3C based client (web2driver(webdriverio as its backend)) for Appium 2.0.
Could you confirm if this inspector works in your environment?
I guess vendors which modify `desiredCapabilities` in https://github.com/appium/appium-inspector/blob/d419cd71af9de62e8f8d1931ddad1650cc4acce1/app/renderer/actions/Session.js#L199 should check carefully..

The below mention is based on relevant contributors related to each vendor in Appium-Desktop.
Sorry if this mention reached someone who already no longer belonged to the company...

- Perfecto: @Dan-Maor @UziEilon
- BrowserStack: @huzaifaiftikhar
- BitBar: @Walther
- Kobiton: @khanhdodang
- pCloudy: @shibupanda
- TestingBot: @testingbot
- ExperiTest: @ArielExperitest
- Roboticmobi: @donmezburak


cc @wswebcreation (I know you already checked in another issue. Thanks so much!)

**Comment by KazuCocoa** [comment]
note: https://github.com/appium/appium-inspector/pull/118 is a known issue.

**Comment by huzaifaiftikhar** [comment]
@KazuCocoa Thanks for mentioning. Should we check with the latest master code or the latest release of appium-inspector i.e. [2021.8.5](https://github.com/appium/appium-inspector/releases/tag/v2021.8.5)?

**Comment by KazuCocoa** [comment]
I think #118 branch should try first. (although potentially the same issue addressed in the PR happen)

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-inspector#development is how to launch the inspector instance

**Comment by testingbot** [comment]
Thanks for letting us know. We've sent a PR with a fix for TestingBot integration: https://github.com/appium/appium-inspector/pull/130

**Comment by Dan-Maor** [comment]
Thanks for the heads up, we will start working on it as soon as possible.

**Comment by huzaifaiftikhar** [comment]
@KazuCocoa I've created a PR for fixing the integration with BrowserStack. Can you please review and merge? https://github.com/appium/appium-inspector/pull/139

**Comment by wswebcreation** [comment]
@KazuCocoa and @jlipps ,

Sauce now works with this version
![image](https://user-images.githubusercontent.com/11979740/134119098-1adbc59b-dd31-4f63-a7cf-c97b6f71d37e.png)
thanks!

We now need to fix accepting W3C caps on real devices which we didn't supported yet. If issues come to this repo then please close them and or ping me in them

**Comment by vishalweb** [comment]
Bitbar Team @Walther , can you comment on this issue where we are not able to connect to bitbar device using Appium inspector. I had raised issue 123 which has been closed and moved to this issue. As per discussion with Appium Team, it's an issue at your end.

**Comment by PatrickFlaherty** [comment]
I created a ticket with Bitbar around this issue and they said they would deploy a fix today the 29th. I just tried it and indeed it is fixed.

**Comment by vishalweb** [comment]
Hi Patrick, Thanks for the information. however I am still not able to connect to bitbar device . I am getting following error. Would you mind sharing if you are passing any additional capabilities to connect to the device.

<img width="1154" alt="image" src="https://user-images.githubusercontent.com/8010956/135389019-fb3a0ad1-04ce-44eb-85ae-53cacd8d486d.png">

**Comment by jlipps** [comment]
I checked the Inspector code, and indeed the `testdroid_source` and `testdroid_apiKey` capabilities are not prefixed with a valid W3C vendor prefix. So someone at bitbar should make a PR to this repo to update them with the appropriate W3C-valid capability names (maybe they should be `appium:testdroid_source` etc?)

**Comment by KazuCocoa** [comment]
I think ExperiTest and BitBar haven't responded here.
Let me see if they have support email addresses for non-account holders and send this issue to them.

**Comment by PatrickFlaherty** [comment]
@vishalweb, I was experiencing an issue with Appium Inspector 1.x and that is what Bitbar fixed. I did see the error your seeing with Inspector 2.0. I suspect the Bitbar possibly does not have support for Appium 2.0 but that just my guess.

**Comment by KazuCocoa** [comment]
Ok, I've sent a support email to BitBar and ExperiTest.

**Comment by vishalweb** [comment]
Thanks @PatrickFlaherty , @KazuCocoa and  @jlipps . Will check with Appium Inspector 1.x version and wait for update from Bitbar on support ticket raised

**Comment by lastverb** [comment]
https://github.com/appium/appium-inspector/issues/128#issuecomment-930779223
> Note: passing the id generated on uploading ipa in bitbar_app and app package in app capability, removed from below image
This error is not a Bitbar response. Library used in this project validates capabilities before sending and this is the result. Example:
<img width="1268" alt="Screenshot 2021-10-05 at 15 50 02" src="https://user-images.githubusercontent.com/2544298/136182235-5b28adb4-170b-4447-bb61-4f89533913db.png">
Those caps are added here https://github.com/appium/appium-inspector/blob/main/app/renderer/actions/Session.js#L312 . I have no idea what bitbar_source capability is. Bitbar doesn’t read or use it.

Bitbar accepts many formats for same capability regarding custom Bitbar capabilities. Including ones not in W3C spec with too long backward compatibility. Assuming there is capability named ‘name’:
1. `name`
2. `bitbar_name`
3. `appium:bitbar_name`
4. `bitbar:name`
5. Sending any of above inside `bitbar:options` object either inside or on the same level as `alwaysMatch`
6. few others that should not be pointed to

Options 1. and 2. are blocked by library used in this project. It’s most likely possible to disable this validation. Not all libraries do this.
Option 3. is what is currently being done when providing `bitbar_name` like before and checking prefix checkbox. Except those caps provided by application, not the user, don’t have prefixes.
Options 4. and 5. are preferred and most fitting W3C spec, however they only work for bitbar custom caps and you cannot pass standard appium nor webdriver capabilities this way. The special case is `app` capability. Original appium `app` requires path to application file and bitbar custom capability `app` expects id for app uploaded to bitbar. Probably because of some bug `app` alone (option 1) inside `bitbar:options` won’t work.

If this application is going to differentiate between vendor specific caps and appium ones prefixing both with separate prefixes, here is a list of bitbar specific capabilities:
`username
password
apiKey
target
project
description
testrun
device
app
locale
testTimeout
testrunId
findDevice
frameworkId
appiumVersion
multiSessionWait
screenResolution
idleTimeout
testName
timezone
recordVideo`

**Comment by KazuCocoa** [comment]
Like other vendors did in https://github.com/appium/appium-inspector/issues/128#issuecomment-913809011, we still need BitBar's help to update their capabilities (probably only: https://github.com/appium/appium-inspector/blob/6dd350af4228f0e24df139f35978151a8b0f4ea8/app/renderer/actions/Session.js#L312-L313) for W3C spec.

The reason why _embedded Appium inspector in Appium desktop v1.20_ worked was the old inspector followed non-W3C spec capabilities which would no longer work in Appium 2.0. This inspector is also going to follow W3C WebDriver spec. Vendors in https://github.com/appium/appium-inspector/issues/128#issuecomment-913809011 have updated their capabilities in this inspector to follow W3C spec. (They have added their own vendor prefix in their own capabilities.)

**Comment by vishalweb** [comment]
Thanks @KazuCocoa 
Bitbar Team  @Walther, would you be able to suggest if this fix is expected soon from Bitbar .

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-inspector/releases/tag/v2021.12.2 has most of fixes.
https://github.com/appium/appium-inspector/pull/225 is remaining, but let me check it on the pr itself.
